modelling/src/main/java/org/axonframework/modelling/configuration/StatefulCommandHandlingModule.java

```diff
@@ -227,6 +230,30 @@ CommandHandlerPhase commandHandler(
         CommandHandlerPhase commandHandlingComponent(
                 @Nonnull ComponentFactory<CommandHandlingComponent> handlingComponentBuilder
         );
+
+        /**
+         * Registers the given {@code handlingComponentBuilder} as an {@link AnnotatedCommandHandlingComponent} within
+         * this module.
+         * <p>
+         * This will scan the given {@code handlingComponentBuilder} for methods annotated with {@link CommandHandler}
+         * and register them as command handlers for the {@link org.axonframework.commandhandling.CommandBus} of the
+         * {@link org.axonframework.configuration.ApplicationConfigurer}.
+         *
+         * @param handlingComponentBuilder A factory method of a {@link CommandHandlingComponent}. Provides the
+         *                                 {@link org.axonframework.configuration.NewConfiguration} to retrieve
+         *                                 components from to use during construction of the command handling
+         *                                 component.
+         * @return The command handler phase of this builder, for a fluent API.
+         */
+        default CommandHandlerPhase annotatedCommandHandlingComponent(
```

Comment:
I like the addition, but I need to start a conversation about it too. Do we want the "generic" `StatefulCommandHandlingModule` to have anything to do with annotations?

It's that question I was wondering myself earlier on wit the `EntityBuilders` too. Hence why I made an `EventSourcedEntityBuilder` and `AnnotatedEventSourcedEntityBuilder`.

To be frank, not sure if we should pull that idea to this level too, but wanted to ensure this paradigm shift doesn't only live in my mind. Hence the comment.

Reply:
I understand where you are coming from, and maybe you are right. 
But, it would mean users need to guess about the different `CommandHandlingComponent` implementations to choose from :thinking:.
Maybe we discuss it in the meeting tomorrow?